src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAccountManager.java

```diff
@@ -1999,4 +1999,49 @@ public Map<SchoolInfoStatus, Long> getSchoolInfoStats() throws SegueDatabaseExce
   public Long getNumberOfAnonymousUsers() throws SegueDatabaseException {
     return temporaryUserCache.getCountOfAnonymousUsers();
   }
+
+  public RegisteredUserDTO updateTeacherPendingFlag(final Long userId, boolean newFlagValue)
+      throws SegueDatabaseException, NoUserException {
+    RegisteredUser user = findUserById(userId);
+    if (user == null) {
+      throw new NoUserException("No user found with this ID.");
+    }
+    user.setTeacherPending(newFlagValue);
+    RegisteredUser updatedUser = database.createOrUpdateUser(user);
+    return dtoMapper.map(updatedUser, RegisteredUserDTO.class);
+  }
+
+  public void sendRoleChangeRequestEmail(final HttpServletRequest request, final RegisteredUserDTO currentUser,
+                                         final Role requestedRole, final Map<String, String> requestDetails)
+      throws SegueDatabaseException, ContentManagerException {
+    String roleName = requestedRole.toString();
+    String userSchool = Objects.requireNonNullElse(currentUser.getSchoolId(), currentUser.getSchoolOther());
+    String verificationDetails = requestDetails.get("verificationDetails");
+    String otherInformation = Objects.requireNonNullElse(requestDetails.get("otherInformation"), "");
+    String emailSubject = String.format("%s Account Request", roleName);
+    String emailMessage = String.format(
+        "Hello,\n<br>\n<br>"
+            + "Please could you convert my Isaac account into a teacher account.\n<br>\n<br>"
+            + "My school is: %s\n<br>"
+            + "A link to my school website with a staff list showing my name and email"
+            + " (or a phone number to contact the school) is: %s\n<br>\n<br>\n<br>"
+            + "Any other information: %s\n<br>\n<br>"
+            + "Thanks, \n<br>\n<br>%s %s",
+        userSchool, verificationDetails, otherInformation, currentUser.getGivenName(), currentUser.getFamilyName());
+    emailManager.sendContactUsFormEmail(properties.getProperty(Constants.MAIL_RECEIVERS),
+        new ImmutableMap.Builder<String, Object>()
+            .put("contactGivenName", currentUser.getGivenName())
+            .put("contactFamilyName", currentUser.getFamilyName())
+            .put("contactUserId", currentUser.getId())
+            .put("contactUserRole", currentUser.getRole())
+            .put("contactEmail", currentUser.getEmail())
+            .put("contactSubject", emailSubject)
+            .put("contactMessage", emailMessage)
+            .put("replyToName", String.format("%s %s", currentUser.getGivenName(), currentUser.getFamilyName()))
+            .build());
+
+    logManager.logEvent(currentUser, request, SegueServerLogType.CONTACT_US_FORM_USED,
```

Comment:
I'm not saying that this is wrong, but is it necessary to have the request information here? It looks like the only reason you ask for the request in this method is to log it. There are other methods in this log manager that allow you to log the rest of the things you want to log, without the request. So my question is: why is it being done like this? I realise that this is also done this way in other places, but why here? And again, I'm not saying it's wrong, I'm just trying to understand it.

Reply:
I just copied the behaviour of the contactUs endpoint used in the current process and wasn't sure whether the log information was reviewed by STEM or not. If we don't need that information for the dedicated endpoint this logging call can be revised. 🤷

Reply:
I have no idea, I don't know where this logs end or who is looking at them. But it would be good to spend some time trying to figure it out. That way we can make a decision. Otherwise, if we just copy things without understanding them, we are at risk of making mistakes, or adding unnecessary code and/or log data.